### DISCUSSION TITLE
PHP Fatal Error When Running Data Center Asset Report

### DISCUSSION THREAD

**Discussion by ghost** [opendcim/openDCIM#1524] [Q&A]
Running OpenDCIM version 20.02 on RHEL 7.9. 

When we attempt to run the report Data Center Asset Report (Excel) the report gets to 90% completion on the dial and as the status switches from "Formatting Spreadsheet" to "Preparing to Transmit" we get the following error in the httpd error log

 PHP Fatal error:  Allowed memory size of 880803840 bytes exhausted (tried to allocate 85 bytes) in /opt/openDCIM/openDCIM-20.02/vendor/phpoffice/phpexcel/Classes/PHPExcel/CachedObjectStorage/CacheBase.php on line 177, referer: https://<mysite>/report_asset_Excel.php
 
 Tried updating the php.ini memory limits from 128M to 256M, to 512M, to 1024M, and even -1 (I know, a no no) but each time we go to run the report it fails at 90%. The Data Center Asset Report (non Excel) runs just fine and all other reports seem to complete just fine. I also did a mysqlcheck on the dcim database and all of the tables came back as OK. 
 
 Looked through other logs within httpd and /var/log itself and nothing that seems to stand out. Anyone have any additional ideas or ran into something similar in the past?

**Comment by cpbonamico** [comment]
Hello, Yes..ran into this exact issue. Edit the "report_asset_Excel.php" file and look at line 151 and change the amount of memory being allocated to this report. 
<img width="400" alt="report_asset_Excel" src="https://github.com/opendcim/openDCIM/assets/78433633/2149b0a7-b957-4c39-81b2-c0828633182e">

**Reply by jdawger765** [reply]
That was indeed the fix. Thank you for the reply
